src/it/java/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/ClassAnnotationsTest.java

```diff
@@ -38,31 +35,12 @@ protected String getPackageLocation() {
 
     @Test
     public void testAnnotation() throws Exception {
-        final Class<AnnotationLocationCheck> clazz1 = AnnotationLocationCheck.class;
-        final Class<InvalidJavadocPositionCheck> clazz2 = InvalidJavadocPositionCheck.class;
-
-        final Configuration annotationConfig = getModuleConfig("AnnotationLocation",
-                "AnnotationLocationMostCases");
-        final Configuration invalidJavadocPositionConfig =
-                getModuleConfig("InvalidJavadocPosition");
-        final List<Configuration> childrenConfigs = new ArrayList<>();
-        childrenConfigs.add(annotationConfig);
-        childrenConfigs.add(invalidJavadocPositionConfig);
-        final Configuration treeWalkerConfig = createTreeWalkerConfig(childrenConfigs);
-
-        final String annotationAloneMsg = "annotation.location.alone";
-        final String invalidJavadocPositionMsg = "invalid.position";
-
-        final String[] expected = {
-            "17:5: " + getCheckMessage(clazz2, invalidJavadocPositionMsg),
-            "25:22: " + getCheckMessage(clazz1, annotationAloneMsg, "SomeAnnotation2"),
-            "28:22: " + getCheckMessage(clazz1, annotationAloneMsg, "SomeAnnotation2"),
-            "29:5: " + getCheckMessage(clazz2, invalidJavadocPositionMsg),
-        };
+        final Map<String, String[]> listOfModules = new HashMap<>();
+        listOfModules.put("AnnotationLocation", new String[] {"AnnotationLocationMostCases"});
+        listOfModules.put("InvalidJavadocPosition", null);
```

Comment:
```
        listOfModules.put("AnnotationLocationMostCases");
        listOfModules.put("InvalidJavadocPosition");
```

id and name of module should be unique in config.
so in handleCheckerChildren you try to load by class name first, if there is exception, you should load by id.

Reply:
Done. 

https://github.com/checkstyle/checkstyle/blob/a0f82691156c959fbb7dccd991c3e5c4f83022e1/src/it/java/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/ClassAnnotationsTest.java#L35-L42